mirrord/agent/src/runtime.rs

```diff
@@ -171,30 +214,55 @@ impl ContainerdContainer {
             },
         }
     }
+
+    async fn get_task_client(&self) -> Result<TasksClient<Channel>> {
+        let channel = self.get_channel().await?;
+        Ok(TasksClient::new(channel))
+    }
+
+    async fn get_container_client(&self) -> Result<ContainersClient<Channel>> {
+        let channel = self.get_channel().await?;
+        Ok(ContainersClient::new(channel))
+    }
 }
 
 #[async_trait]
 impl ContainerRuntime for ContainerdContainer {
-    async fn get_pid(&self) -> Result<u64> {
-        let mut client = self.get_client().await?;
+    async fn get_info(&self) -> Result<ContainerInfo> {
+        let mut client = self.get_task_client().await?;
         let container_id = self.container_id.to_string();
         let request = GetRequest {
             container_id,
             ..Default::default()
         };
         let request = with_namespace!(request, DEFAULT_CONTAINERD_NAMESPACE);
-        let response = client.get(request).await?;
+        let response = client.get(request).await?.into_inner();
         let pid = response
```

Comment:
```suggestion
        let pid = client.get(request).await?.into_inner()
```
I think we don't need to keep `response` as an intermediate value here, right? but this is nitpicky and completely optional.